### PR TITLE
Create action for GitHub release

### PR DESCRIPTION
**Pull request by Smaug123**
Rationale is in the "Why?" section of the README.

Observe the [successful run](https://github.com/Smaug123/test-repo/actions/runs/12176186326/job/33961396316) against [a test PR](https://github.com/Smaug123/test-repo/pull/6/commits/35f11423e8de1e8ebcbd4a5dfd196da15cf53030). This produced the expected releases.

The "upload binary data" is tested: https://github.com/Smaug123/test-repo/releases/tag/tmp.xAUcJ5OU is one such release, constructed from https://github.com/Smaug123/test-repo/pull/6 at commit 35f11423e8de1e8ebcbd4a5dfd196da15cf53030 .

### CODE REVIEW

File: github-release/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+## `tag`
+
+The tag name to which the GitHub release will correspond.
+This is used both as the release name and as the tag to which the release points.
+
+If this doesn't exist at the time this step runs, the tag is created to point to `target-commitish`.
+
+## `github-token`
+
+A GitHub token with at least `contents: "write"` perms, and optionally also `actions: "write"` (though GitHub does not document the conditions under which this is required).
```
Comment (jgiannuzzi) [inline github-release/README.md:44]: IIRC `actions: "write"` is required when a push is made that includes changes to `.github/workflows/*.yaml`.

**Review by jgiannuzzi — DISMISSED** [review]
LGTM

File: github-release/README.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ github-release:
         generate-release-notes: true
         draft: true
         prerelease: true
+        binary-contents: path/to/binary
```
Comment (jgiannuzzi) [inline github-release/README.md]: we probably want this to be a list paths to assets to upload?

Comment (jgiannuzzi) [reply github-release/README.md]: (in GHA, you would typically use EOL-separated strings for that)

Comment (Smaug123) [reply github-release/README.md]: Oh my *god* the `jq` in the `ubuntu-latest` image is *ancient*. Why do they make everything so hard 😭

**Comment by Smaug123** [comment]
Raised https://github.com/G-Research/common-actions/pull/7 as an alternative to this PR.

**Comment by Smaug123** [comment]
Ready for review again.

**Review by jgiannuzzi — APPROVED** [review]
LGTM

**Pull request merged**